src/AppBundle/Controller/Report/ReportController.php

```diff
@@ -401,6 +437,94 @@ public function update(Request $request, $id)
+            $report->setProfDeputyCostsHasPrevious($data['prof_deputy_costs_has_previous']);
+            foreach ($report->getProfDeputyPreviousCosts() as $pc) {
+                $this->getEntityManager()->remove($pc);
+            }
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
+        }
+
+        if (!empty($data['prof_deputy_costs_has_interim']) && $data['prof_deputy_costs_has_interim']) {
+            $report->setProfDeputyCostsHasInterim($data['prof_deputy_costs_has_interim']);
+            // remove interim if changed to "no"
+            if ($data['prof_deputy_costs_has_interim'] === 'no') {
+                foreach ($report->getProfDeputyInterimCosts() as $ic) {
+                    $this->getEntityManager()->remove($ic);
+                }
+            } else if ($data['prof_deputy_costs_has_interim'] === 'yes') {
+                $report->setProfDeputyFixedCost(null);
+            }
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
+        }
+
+        if (array_key_exists('prof_deputy_interim_costs', $data)) {
+            // wipe existing interim costs in order to overwrite
+            // TODO consider keeping and updating the existing ones if simpler to implement
+            foreach ($report->getProfDeputyInterimCosts() as $ic) {
+                $this->getEntityManager()->remove($ic);
+            }
+            // add new
+            foreach ($data['prof_deputy_interim_costs'] as $row) {
+                if ($row['date'] && $row['amount']) {
+                    $report->addProfDeputyInterimCosts(
+                        new EntityDir\Report\ProfDeputyInterimCost($report, new \DateTime($row['date']), $row['amount'])
+                    );
+                }
+                if (count($report->getProfDeputyInterimCosts())) {
+                    $report->setProfDeputyCostsHasInterim('yes');
+                }
+            }
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
+            $this->getEntityManager()->flush();
+        }
+
+        if (array_key_exists('prof_deputy_fixed_cost', $data)) {
+            $report->setProfDeputyFixedCost($data['prof_deputy_fixed_cost']);
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
+        }
+
+        if (array_key_exists('prof_deputy_costs_amount_to_scco', $data)) {
+            $report->setProfDeputyCostsAmountToScco($data['prof_deputy_costs_amount_to_scco']);
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
+        }
+
+        if (array_key_exists('prof_deputy_costs_reason_beyond_estimate', $data)) {
+            $report->setProfDeputyCostsReasonBeyondEstimate($data['prof_deputy_costs_reason_beyond_estimate']);
+            $report->updateSectionsStatusCache([Report::SECTION_PROF_DEPUTY_COSTS]);
```

Comment:
Possibly move this to a single call towards the end of the method since cache status changing doesnt matter part way though the method

Reply:
I agree. It's probably just a safety mechanism to always update status cache as soon as a property is set on the entity which may alter the status of that section, just in case a new line is added afterwards which relies on the current status.

Reply:
will implement